examples/z_pub_attachment.c

```diff
@@ -80,8 +78,7 @@ int main(int argc, char** argv) {
     z_publisher_put_options_default(&options);
 
     // allocate attachment data
-    kv_pair_t kvs[2];
-    kvs[0] = (kv_pair_t){.key = "source", .value = "C"};
+    kv_pair_t kvs[2] = {(kv_pair_t){.key = "source", .value = "C"}};
```

Comment:
IMHO original way was more clear (because of  the similar assignment in the loop)